app/services/survey_monkey_parser.rb

```diff
@@ -96,12 +129,23 @@ def score_options
       score_question.dig('details', 'answers', 'choices')
     end
 
-    def score_answer_given
-      score_question['answers'].first['choice_id']
+    def score_answers_given
+      score_question['answers']
+    end
+
+    def parsed_score_answer(choice_id)
+      choice = score_options.find { |c| c['id'] == choice_id }
+
+      if survey.use_weights && choice['weight'].present?
+        choice['weight']
+      else
+        choice['text'].gsub(/[^0-9]/, '').to_i
+      end
     end
 
-    def parsed_score_answer
-      score_options.find { |c| c['id'] == score_answer_given }['text'].gsub(/[^0-9]/, '').to_i
+    def average_score
+      total = score_answers_given.reduce(0) { |sum, answer| sum + parsed_score_answer(answer["choice_id"]) }
+      total / (score_answers_given.size.nonzero? || 1)
```

Comment:
`score_answers_given.size.nonzero?` return true / false? If yes this code is incorrect

Reply:
It returns `nil` or `score_answers_given.size`